processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -137,7 +176,11 @@ public boolean remove(String lookupName)
   {
     final LookupExtractorFactory lookupExtractorFactory = lookupMap.remove(lookupName);
     if (lookupExtractorFactory != null) {
-      LOGGER.debug("Removing lookup [%s]", lookupName);
+      LOGGER.debug("Removed lookup [%s]", lookupName);
+      if (!isClosed() && lookupSnapshotTaker != null) {
+        // if it is closed the final snapshot will be taken at stop() method
+        lookupSnapshotTaker.takeSnapshot(getAllAsList());
```

Comment:
Sometimes this is protected by the start/stop lock and sometimes it is not. It seems that it should probably be protected  here. It is possible to pass `isClosed`, but have `stop` remove elements between `isClosed` and `getAllAsList`